superset/result_set.py

```diff
@@ -70,9 +70,14 @@ def stringify_values(array: NDArray[Any]) -> NDArray[Any]:
         for obj in it:
             if na_obj := pd.isna(obj):
                 # pandas <NA> type cannot be converted to string
-                obj[na_obj] = None  # type: ignore
+                obj[na_obj] = None
             else:
-                obj[...] = stringify(obj)  # type: ignore
+                try:
+                    # for simple string conversions
+                    # this handles odd character types better
+                    obj[...] = obj.astype(str)
```

Comment:
@eschutho There are some failed unit tests comparing stringified objects.  It seems like the 
`obj.astype(str)` call may return inverted use of single and double quotes from `stringify(obj)`
example from failed tests:
returned: 
`"{'chart_name': 'scatter'}"` <-- not technically valid JSON
and 
Expected: 
`'{"chart_name": "scatter"}'` <-- valid JSON

Not sure if this matters in actual python code execution, but from a JSON point of view the new stringified output is not valid to use in JSON.parse() and would throw JS error

Regardless of if it matters if there is valid JSON here, the unit tests that were in place. before this change are in the valid JSON format and are failing because of the quote inversion. 

https://github.com/apache/superset/blob/d954c3df8604dc4b6a2459a17dd39450a1d57638/tests/integration_tests/result_set_tests.py#L165

Reply:
I think it should be fine, because even though we're using `json.dumps()` in `stringify` we're not trying to generate valid JSON, we're just trying to make sure complex/nested objects can be converted to a string. So it should be fine to update the tests to match the new output.